### ISSUE TITLE
Clarification on PMIx_Spawn info attributes

### ISSUE THREAD

**Issue by naughtont3** [pmix/pmix-standard#400]
**Describe your question or clarification request here**

The specification for `PMIx_Spawn()` and `PMIx_Spawn_nb()` include a list of attributes that can be passed via the info key/values.  The below text indicates these `pmix_info_t` items may be passed via the `job_info` or `app` arguments.  

Should this be clarified to indicate what info items are valid to be passed via the `job_info` or `app` arg (or both)?

### Suggested Clarification

In practice, when  using PRRTE-2.0.2 and OpenPMIx-4.1.2, we found that some attributes (e.g., `PMIX_MAPBY`) needed to be in the `job_info` area of spawn.  It is unclear if this is purely an implementation detail or something that should be clarified in the specification.

### References

Please list specific chapters/sections in the PMIx Standard related to your query.

Text from PMIx-4.0 spec under PMIx_Spawn description (Page 159, Lines 32-33)
```
... In addition, they are required to support the following attributes when present in either the job_info or the info array of an element of the apps array:
```

**Comment by rhc54** [comment]
> In practice, when using PRRTE-2.0.2 and OpenPMIx-4.1.2, we found that some attributes (e.g., PMIX_MAPBY) needed to be in the job_info area of spawn. It is unclear if this is purely an implementation detail or something that should be clarified in the specification.

I noticed the `map-by` constraint myself the other day. Currently, PRRTE doesn't support different mapping strategies per app in the same job. It would be a pretty significant effort to change that, and I'm not sure if it would be worth the effort.

Do you have a use-case that requires different mapping strategies for apps in the same job?

**Comment by naughtont3** [comment]
No, we don't have a use case.  It was more a matter of a snafu when using it and not knowing exactly where to put the info key.  I agree it might be kind of difficult to do per-app mapping strategies.  I was thinking we might want to clarify, maybe as an advise to user, that currently the following XYZ attributes are only supported at the overall job level.

**Comment by rhc54** [comment]
Yeah, probably should just clarify better what goes into each of the two arrays, and what can go into either/both of them.

**Comment by naughtont3** [comment]
OK, i'll try to take a swing at that text.  Main question will be formatting and identifying what attrs below in which array or both.

**Comment by jjhursey** [comment]
@naughtont3 have you had a chance to make a pass at some text here, or do you need help?

**Comment by naughtont3** [comment]
I took a quick look last week.  I think the main issue is that it is unclear how we want to annotate items that should be in (J)ob or (A)pp, or both.  I do not think we want to add another listing, so maybe add a flag or marker next to the attribute (e.g., (J), (A), (J,A) to indicate only Job, only App arrary or appear in both/either, respectively.
